### PR TITLE
Current grade support

### LINKED ISSUE #24: support current grade

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
fixes #24 

#### What's this PR do?
introduces support for the edX current grade rest API introduced with PR https://github.com/edx/edx-platform/pull/13285

#### Where should the reviewer start?
`edx_api/grades/__init__.py`

#### How should this be manually tested?
run integration tests: note that the ccx integration test will fail (because of a well known bug on edx)

#### Any background context you want to provide?
the edx pr is still under review, even if the overall structure should not change

### CODE REVIEW

**Comment by pdpinch** [comment]
@amir-qayyum-khan can you review this PR?

**Comment by amir-qayyum-khan** [comment]
@giocalitri functionality looks good, please address my above comments

**Comment by giocalitri** [comment]
@amir-qayyum-khan I do not see any comment

**Comment by amir-qayyum-khan** [comment]
oops new git UX. Sorry, i posted those comments 4 hrs ago, they were in pending state. now i press commit to sent (there was another button to make them go ;( )

**Comment by giocalitri** [comment]
no problem :-)

File: edx_api/grades/__init__.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    def __init__(self, requester, base_url):
+        """
+        Args:
+            requester (Requester): an authenticated objects for requests to edX
+            base_url (str): string representing the base URL of an edX LMS instance
+        """
+        self.requester = requester
+        self.base_url = base_url
+
+    def get_student_current_grade(self, username, course_id):
```
Comment (giocalitri) [inline edx_api/grades/__init__.py]: we do not do validation about the structure of the data in this library: we rely on the fact that edX will handle errors properly with proper HTTP statuses

File: edx_api/grades/__init__.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    course_key=course_id
+                )
+            )
+        )
+
+        resp.raise_for_status()
+
+        return CurrentGrade(resp.json()[0])
+
+    def get_student_current_grades(self, username, course_ids=None):
```
Comment (giocalitri) [inline edx_api/grades/__init__.py]: I do not think it's necessary: a `None` username will result in a request to edX containing the string "None" as username that will likely fail.

File: edx_api/grades/__init__.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        """
+        Returns an Certificates object with the user certificates.
```
Comment (giocalitri) [inline edx_api/grades/__init__.py]: sure: will fix

**Comment by giocalitri** [comment]
Done

**Comment by amir-qayyum-khan** [comment]
👍 💯

**Comment by amir-qayyum-khan** [comment]
@giocalitri by the way what are you thought on my suggestions?
what if used enters invalid username, course_ids=None? will call api crash or not?

**Comment by giocalitri** [comment]
the only place where it matters is in `get_student_curret_grade` (the other handles the `None`)
in this case this code
```
'/api/grades/v0/course_grade/{course_key}/users/?username={username}'.format(
 +                    username=username,
 +                    course_key=course_id
 +                )
```

will result in a get request to the URL
`/api/grades/v0/course_grade/None/users/?username=None`
that will result in a 404 from edx because `"None"` is not a valid course key string

**Comment by amir-qayyum-khan** [comment]
👍 thanks

**Pull request merged**